module/VuFind/src/VuFind/Log/Logger.php

```diff
@@ -51,110 +52,227 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Site
  */
-class Logger extends BaseLogger
+
+
+class Logger implements LoggerInterface
 {
     /**
      * Is debug logging enabled?
      *
-     * @var bool
+     * @var bool`
      */
-    protected $debugNeeded = false;
+    protected bool $debugNeeded = false;
 
     /**
      * User IP address reader
      *
      * @var UserIpReader
      */
-    protected $userIpReader;
+    protected UserIpReader $userIpReader;
+    protected MonologLogger $monologLogger;
+
+    protected const LEVEL_MAP = [
+        'err'       => LogLevel::ERROR,
+        'warn'      => LogLevel::WARNING,
+    ];
+
+    public function __construct(UserIpReader $userIpReader, ?MonologLogger $monologLogger)
+    {
+        $this->userIpReader = $userIpReader;
+        $this->monologLogger = $monologLogger ?? new MonologLogger('default');
+    }
 
     /**
-     * Constructor
+     * System is unusable.
      *
-     * Set options for a logger. Accepted options are:
-     * - writers: array of writers to add to this logger
-     * - exceptionhandler: if true register this logger as exceptionhandler
-     * - errorhandler: if true register this logger as errorhandler
+     * @param mixed[] $context
+     */
+    public function emergency(string|\Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::EMERGENCY, $message, $context);
+    }
+
+    /**
+     * System is unusable.
+     *
+     * @param mixed[] $context
+     */
```

Comment:
Should we add a `@deprecated` annotation to these backward-compatibility methods to encourage people to use the PSR versions, or do you think it's best to support everything in parallel?